core/src/main/java/org/elasticsearch/common/transport/NetworkExceptionHelper.java

```diff
@@ -39,10 +39,13 @@ public static boolean isCloseConnectionException(Throwable e) {
             return true;
         }
         if (e.getMessage() != null) {
-            // UGLY!, this exception messages seems to represent closed connection 
+            // UGLY!, this exception messages seems to represent closed connection
             if (e.getMessage().contains("Connection reset by peer")) {
                 return true;
             }
+            if (e.getMessage().contains("Connection reset")) {
```

Comment:
Can you just remove the "by peer" from the first, since matching this new case is a superset of matching the first case?